silk-react-components/src/HierarchicalMapping/Components/MappingRule/Forms/ValueMappingRuleForm.jsx

```diff
@@ -96,138 +100,128 @@ const ValueMappingRuleForm = React.createClass({
                     }
                 );
         } else {
-            hierarchicalMappingChannel
-                .subject('ruleView.change')
-                .onNext({id: 0});
             this.setState({
-                create: true,
-                loading: false,
-                type: MAPPING_RULE_TYPE_DIRECT,
-                valueType: { nodeType: 'StringValueType' },
-                sourceProperty: '',
-                isAttribute: false,
-                initialValues: {},
+                loading: false
             });
+            EventEmitter.emit(MESSAGES.RULE_VIEW.CHANGE, {id: 0});
         }
-    },
-    handleConfirm(event) {
+    }
+    
+    handleConfirm = (event) => {
         event.stopPropagation();
         event.persist();
```

Comment:
the event is persisted to run `event.stopPropagation()` again within `this.handleClose(event)` which seems some kind of duplicated to me.
Is the `event.stopPropagation()` needed at all? Afaik there is no other onClick event trigger in range so it maybe can dropped at all (on both methods)

Reply:
I removed the `stopPropagation` from `handleClose`